packages/flutter/lib/src/material/chip.dart

```diff
@@ -103,81 +111,658 @@ class Chip extends StatelessWidget {
+      }
+    }
+
+    add(avatar, 'avatar');
+    add(label, 'label');
+    add(deleteIcon, 'deleteIcon');
+    add(container, 'container');
+    return value;
+  }
+
+  @override
+  bool get sizedByParent => false;
+
+  static double _minWidth(RenderBox box, double height) {
+    return box == null ? 0.0 : box.getMinIntrinsicWidth(height);
+  }
+
+  static double _maxWidth(RenderBox box, double height) {
+    return box == null ? 0.0 : box.getMaxIntrinsicWidth(height);
+  }
+
+  static double _minHeight(RenderBox box, double width) {
+    return box == null ? 0.0 : box.getMinIntrinsicWidth(width);
+  }
+
+  static Size _boxSize(RenderBox box) => box == null ? Size.zero : box.size;
+
+  static BoxParentData _boxParentData(RenderBox box) => box.parentData;
+
+  @override
+  double computeMinIntrinsicWidth(double height) {
+    // The overall padding isn't affected by missing avatar or delete icon
+    // because we add the padding regardless to give extra padding for the label
+    // when they're missing.
+    final double overallPadding = theme.labelPadding.horizontal + _iconPadding.horizontal * 2.0;
+    return overallPadding + _minWidth(avatar, height) + _minWidth(label, height)
+        + _minWidth(deleteIcon, height);
+  }
+
+  @override
+  double computeMaxIntrinsicWidth(double height) {
+    // The overall padding isn't affected by missing avatar or delete icon
+    // because we add the padding regardless to give extra padding for the label
+    // when they're missing.
+    final double overallPadding = theme.labelPadding.horizontal + _iconPadding.horizontal * 2.0;
+    return overallPadding + _maxWidth(avatar, height) + _maxWidth(label, height)
+        + _maxWidth(deleteIcon, height);
+  }
+
+  @override
+  double computeMinIntrinsicHeight(double width) {
+    // This widget is sized to the height of the label only.  The other
+    // widgets are sized to match.
+    return theme.labelPadding.vertical + _minHeight(label, width);
+  }
+
+  @override
+  double computeMaxIntrinsicHeight(double width) => computeMinIntrinsicHeight(width);
+
+  @override
+  double computeDistanceToActualBaseline(TextBaseline baseline) {
+    // The baseline of this widget is the baseline of the label.
+    return label.computeDistanceToActualBaseline(baseline);
+  }
+
+  @override
+  void performLayout() {
+    double iconHeight = _chipHeight;
+    if (label != null) {
+      label.layout(constraints.loosen(), parentUsesSize: true);
+      // Now that we know the height, we can determine how much to shrink the
+      // constraints by for the "real" layout. Ignored if the constraints are
+      // infinite.
+      iconHeight = math.max(iconHeight, _boxSize(label).height);
+      if (constraints.maxWidth.isFinite) {
+        final double allPadding = _iconPadding.horizontal * 2.0 + theme.labelPadding.horizontal;
+        final double iconSizes = (avatar != null ? iconHeight - _iconPadding.horizontal : 0.0)
```

Comment:
Shouldn't this be `iconHeight - _iconPadding.vertical`? Likewise for the next line.

Reply:
Yep (I already fixed that in my next PR, doing it here too).